frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1056,107 +1049,99 @@ class Worksheet extends React.Component {
     // Also, a non-null item could contain a list of bundle_infos, which represent a list of bundles. Usually not all of them need updating.
     // The bundle_infos for bundles that don't need updating are also null.
     // If rawIndexAfterEditMode is defined, this reloadWorksheet is called right after toggling editMode. It should resolve rawIndex to (focusIndex, subFocusIndex) pair.
-    reloadWorksheet = (
+    reloadWorksheet = async (
         partialUpdateItems,
         rawIndexAfterEditMode,
         { moveIndex = false, textDeleted = false } = {},
     ) => {
         if (partialUpdateItems === undefined) {
             $('#update_progress').show();
             this.setState({ updating: true });
-            this.state.ws.fetch({
```

Comment:
There's a lot of frontend changes here...can you comment on what the nature of these changes are with respect to the async loading feature?

Reply:
Mainly, I've refactored `this.state.ws.fetch` to return a promise instead of using callbacks. This way, this simplifies the work done in `componentDidMount()`, since `await this.state.ws.fetch({ brief: true });` can just be called once and then followed by `wait this.state.ws.fetch()`.

Nothing else in terms of frontend functionality was changed.